### PR TITLE
chore: use new monorepo packages

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()
🔍 [Preview Link](https://react-components-git-brkfeat-scripts-monorepo-hashicorp.vercel.app/)

---

## Description

Consume the new packages published from the monorepo that is formerly known as nextjs-scripts. Relevant PR: https://github.com/hashicorp/nextjs-scripts/pull/162

## Release Information

Please select one (**required**)

- [ ] **Major** (This PR introduces a breaking (incompatible API) change)
- [x] **Minor** (This PR adds functionality but it backwards-compatible)
- [ ] **Patch** (This PR adds a backwards-compatible bug fix)

<details>
<summary>Release Notes (<strong>required</strong>)</summary>

</details>

---

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Add release notes to the appropriate section (above).
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Cross Browser Testing](https://crossbrowsertesting.com) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ module.exports = {
     '\\.svg$': 'jest-raw-loader',
   },
   transformIgnorePatterns: [
-    '/node_modules/(?!(@hashicorp/mktg-logos|@hashicorp/react-|@hashicorp/sentinel-embedded|@hashicorp/nextjs-scripts))',
+    '/node_modules/(?!(@hashicorp/mktg-logos|@hashicorp/react-|@hashicorp/sentinel-embedded|@hashicorp/nextjs-scripts||@hashicorp/platform-))',
```
Comment (jescalan) [inline jest.config.js:20]: This is legit right?

```suggestion
    '/node_modules/(?!(@hashicorp/mktg-logos|@hashicorp/react-|@hashicorp/sentinel-embedded|@hashicorp/platform-))',
```

File: next.config.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 const withSwingset = require('swingset')
 
 module.exports = withHashicorp({
   transpileModules: [
     'swingset',
     '@hashicorp/react-.*',
-    '@hashicorp/nextjs-scripts',
+    '@hashicorp/platform-*',
     '@hashicorp/sentinel-embedded',
+    '@hashicorp/platform-product-meta',
```
Comment (jescalan) [inline next.config.js:10]: Shouldnt the glob cover this one?

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "dependencies": {
     "@hashicorp/mktg-global-styles": "^3.0.2",
     "@hashicorp/mktg-logos": "^1.0.2",
-    "@hashicorp/nextjs-scripts": "^17.0.0",
+    "@hashicorp/nextjs-scripts": "^19.0.3",
```
Comment (jescalan) [inline package.json:7]: Can we drop this entirely now?

File: packages/feedback-form/package.json [hunk 4]
```diff
@@ -7,12 +7,12 @@
     "Bryce Kalow"
   ],
   "dependencies": {
+    "@hashicorp/react-button": "^5.0.0",
```
Comment (jescalan) [inline packages/feedback-form/package.json:10]: 👀

**Review by jescalan — APPROVED** [review]
🙌

File: package.json (hunk 3 above)
Comment (BRKalow) [reply package.json:7]: I was running into some issues dropping it completely 😢  [`45c092e` (#256)](https://github.com/hashicorp/react-components/pull/256/commits/45c092e8df48c0e4cd75f2b0183029efbdf05b45) I think we need to get a publish fired off here with all the package deps pointing at the new packages, then that should let us fully drop the scripts dependency.

File: next.config.js (hunk 2 above)
Comment (BRKalow) [reply next.config.js:10]: Ah yeah, good catch.

File: jest.config.js (hunk 1 above)
Comment (BRKalow) [reply jest.config.js:20]: Still need scripts here as its deep in the dependency tree. I think after a publish here things should get buttoned up.

File: packages/feedback-form/package.json (hunk 4 above)
Comment (BRKalow) [reply packages/feedback-form/package.json:10]: _Oops_

File: next.config.js (hunk 2 above)
Comment (BRKalow) [reply next.config.js]: Ended up needing to adjust this slightly so we can handle deps in workspace packages. All seems good now!

File: jest.config.js (hunk 1 above)
Comment (jescalan) [reply jest.config.js]: I think the pattern is still wrong here, there are two pipes before the last item. Maybe this doesn't matter though with the matcher?

**Pull request merged**